modules/mono/csharp_script.cpp

```diff
@@ -952,7 +952,12 @@ void CSharpLanguage::reload_assemblies(bool p_soft_reload) {
 				CRASH_COND(si != nullptr);
 #endif
 				// Re-create script instance
-				obj->set_script(script); // will create the script instance as well
+				obj->set_script(script);
+				// Create the script instance as well, by-passing the tool requirement in set_script
+				ScriptInstance *script_instance = script->instance_create(obj);
```

Comment:
The script could be marked `[Tool]`, which means the `set_script` above could succeed at creating a new instance. As such, we would be creating a second one unnecessarily. This should be inside a `if (obj->get_script_instance() != nullptr)`

Reply:
All right, so the correct solution would be to move the `obj->set_script(script);` line to an `else {` block if the `script_instance` you created is `nullptr`. `set_script_instance` already takes care of setting the script resource as well.

Reply:
Sorry, it seems I edited my comment and wrote the update too late. Your new change works, but it's still creating an unnecessary placeholder instance. The way I describe above should work.